src/io/config.cpp

```diff
@@ -334,8 +334,8 @@ void Config::CheckParamConflict() {
       num_leaves = static_cast<int>(full_num_leaves);
     }
   }
-  // force col-wise for gpu & CUDA
-  if (device_type == std::string("gpu") || device_type == std::string("cuda")) {
+  // force col-wise for gpu
+  if (device_type == std::string("gpu")/* || device_type == std::string("cuda")*/) {
```

Comment:
should we keep device_type==gpu?

Reply:
Let's keep device_type==gpu for a few versions and provide a deprecated warning. For old device_type==cuda, let's remove it after our new CUDA tree learner supports multi-GPU and distributed scenarios. Before that, multi-GPU and distributed scenarios can switch to use the old CUDA tree learner.

Reply:
Ah, do you have plans to remove `gpu` later in the future? I thought OpenCL-based implementation is a competitive advantage over other GBM libraries as it can work on AMD cards and (perhaps) incoming Intel Ark ones.

Reply:
Good suggestion @StrikerRUS. In that case, let's keep the OpenCL version. And can even spend some efforts to further improve it.

Reply:
Agree with @StrikerRUS , I'm in favor of keeping the OpenCL version and using `device_type = cuda` for the CUDA version.